Update draft-dt-tls-external-psk-guidance.md (#32)
tlswg/external-psk-design-team: draft-dt-tls-external-psk-guidance.md

Applications MUST use external PSKs that adhere to the following requirements:

1. Each PSK SHOULD be derived from at least 128 bits of entropy, MUST be at least
128 bits long, and SHOULD be combined with a DH exchange for forward secrecy. As
discussed in {{sec-properties}}, low entropy PSKs, i.e., those derived from less
than 128 bits of entropy, are subject to attack and SHOULD be avoided. Low entropy

> key are only secure against active attack if a PAKE, and not a PSK, is used with TLS.

Reviewer: ```suggestion
keys are only secure against active attack if a Password Authenticated Key Exchange (PAKE) is used with TLS.
```

- Deployments should take care that the length of the PSK identity is sufficient
to avoid obvious collisions.

### PSK Identity Collisions

It is possible, though unlikely, that an external PSK identity may clash with a
resumption PSK identity. The TLS stack implementation and sequencing of PSK callbacks
influences the application's behaviour when identity collisions occur. When a server

> receives a PSK identity in a TLS 1.3 ClientHello, OpenSSL, BoringSSL and mbedTLS 

Reviewer: ```suggestion
receives a PSK identity in a TLS 1.3 ClientHello, some TLS stacks
```

execute the application's registered callback function before checking the stack's
internal session resumption cache. This means that if a PSK identity collision occurs,
the application will be given precedence over how to handle the PSK.

[[TODO.. the application still does not know if an identity collision occured or not.
If we wanted to be 100% explicit, the stack would check its resumption cache and then
callback the application indicating here is a PSK that has a cache hit, please check
your external PSK DB, and let me know how to proceed. Seems like overkill for something

> that should not happen.]]

Reviewer: Yes, this seems like overkill since it's an implementation detail. I'll remove this TODO after merging.

> that should not happen.]]

Reviewer: ```suggestion
```
